File: MoviesExample/core/src/main/java/com/novoda/demo/movies/model/Video.java
package com.novoda.demo.movies.model;

public class Video {

    public String id;
    public String key;
    public String site;
    public String type;

    @Override
    public String toString() {
        return "Video{" +
                "id='" + id + '\'' +
                ", key='" + key + '\'' +
                ", site='" + site + '\'' +
                ", type='" + type + '\'' +
                '}';
    }

    public String trailerUrl() {
        if (!"YouTube".equals(site) || !"Trailer".equals(type)) {
            return null;
        }
        return "https://www.youtube.com/watch?v=" + key;
    }
}

File: MoviesExample/core/src/main/java/com/novoda/demo/movies/model/Video.kt
package com.novoda.demo.movies.model

data class Video(
        val id: String? = null,
        val key: String? = null,
        val site: String? = null,
        val type: String? = null
) {

    fun trailerUrl(): String? {
        return if ("YouTube" != site || "Trailer" != type) {
            null
        } else "https://www.youtube.com/watch?v=" + key!!
    }
}
